test/Dapr.E2E.Test/DaprTestApp.cs

```diff
@@ -31,28 +33,63 @@ public DaprTestApp(ITestOutputHelper output, string appId, bool useAppPort = fal
             this.testOutput = output;
         }
 
-        public (string, string) Start()
+        public string AppId => this.appId;
+
+        public (string httpEndpoint, string grpcEndpoint) Start()
         {
             var (appPort, httpPort, grpcPort, metricsPort) = GetFreePorts();
+
             var componentsPath = Combine(".", "..", "..", "..", "..", "..", "test", "Dapr.E2E.Test", "components");
-            var daprStartCommand = $" run --app-id {appId} --dapr-http-port {httpPort} --dapr-grpc-port {grpcPort} --metrics-port {metricsPort} --components-path {componentsPath}";
             var projectPath = Combine(".", "..", "..", "..", "..", "..", "test", "Dapr.E2E.Test.App", "Dapr.E2E.Test.App.csproj");
-            var daprDotnetCommand = $" -- dotnet run --project {projectPath} --framework {GetTargetFrameworkName()}";
+            var arguments = new List<string>()
+            {
+                // `dapr run` args
+                "run",
+                "--app-id", appId,
+                "--dapr-http-port", httpPort.ToString(CultureInfo.InvariantCulture),
+                "--dapr-grpc-port", grpcPort.ToString(CultureInfo.InvariantCulture),
+                "--metrics-port", metricsPort.ToString(CultureInfo.InvariantCulture),
+                "--components-path", componentsPath,
+                "--log-level", "debug",
+                
+            };
+
+            if (this.useAppPort)
+            {
+                arguments.AddRange(new[]{ "--app-port", appPort.ToString(CultureInfo.InvariantCulture), });
+            }
+
+            arguments.AddRange(new[]
+            {
+                // separator
+                "--",
+
+                // `dotnet run` args
+                "dotnet", "run",
+                "--project", projectPath,
+                "--framework", GetTargetFrameworkName(),
+
+                // Prevent building again, it slows things down as restore can be slow or unreliable in CI environments
+                "--no-build",
+                "--no-restore",
+            });
+
             if (this.useAppPort)
             {
-                daprStartCommand += $" --app-port {appPort}";
-                daprDotnetCommand += $" --urls http://localhost:{appPort}";
+                arguments.AddRange(new[]{ "--urls", $"http://localhost:{appPort.ToString(CultureInfo.InvariantCulture)}", });
             }
-            daprStartCommand += daprDotnetCommand;
 
-            var daprStart = new DaprCommand()
+            // TODO: we don't do any quoting right now because our paths are guaranteed not to contain spaces
+            var daprStart = new DaprCommand(this.testOutput)
             {
                 DaprBinaryName = DaprTestApp.daprBinaryName,
-                Command = daprStartCommand,
+                Command = string.Join(" ", arguments),
                 OutputToMatch = outputToMatchOnStart,
-                Timeout = 10000
+                Timeout = TimeSpan.FromSeconds(30),
```

Comment:
I found that some of these timeouts weren't long enough. it's best to err on the side of really generous timeouts. You only want to hit a timeout if it's really truly not working.

Reply:
Ex: the Java E2E tests use a 60s timeout